lib/puppet/provider/service/freebsd.rb

```diff
@@ -74,7 +74,7 @@ def rc_replace(service, rcvar, yesno)
       if Puppet::FileSystem.exist?(filename)
         s = File.read(filename)
         if s.gsub!(/^(#{rcvar}(_enable)?)=\"?(YES|NO)\"?/, "\\1=\"#{yesno}\"")
-          File.open(filename, File::WRONLY) { |f| f << s }
+          File.open(filename, File::WRONLY | File::TRUNC) { |f| f << s }
```

Comment:
I'd recommend using [`Puppet::FileSystem.replace_file`](https://github.com/puppetlabs/puppet/blob/c3d8122c0203984dd7fbb74dfa1ef73c19950076/lib/puppet/file_system.rb#L402-L417) for that so that the write is done atomically, something like:

```ruby
Puppet::FileSystem.replace_file(path, 0640) { |f| f << s }
```

Reply:
Make sense. I was worried about what permission to set and since we did not managed this before, I assumed we could continue to not manage this :smile: 

I added a commit to take this into account, thanks!